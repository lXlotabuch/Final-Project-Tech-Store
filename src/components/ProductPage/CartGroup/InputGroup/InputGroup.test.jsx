/* eslint-disable react/jsx-props-no-spreading */
import React from 'react'
import { fireEvent, render } from '@testing-library/react'
import InputGroup from './InputGroup'

describe('All test for input group', () => {
  const props = {
    inputHandler: jest.fn(),
    decreaseHandler: jest.fn(),
    increaseHandler: jest.fn(),
    quantity: 10
  }
  const startQuantity = props.quantity

  test('InputGroup shapshot test', () => {
    const { asFragment } = render(<InputGroup {...props} />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('check the props value in the input', () => {
    const { getByDisplayValue } = render(<InputGroup {...props} />)
    expect(getByDisplayValue(startQuantity)).toBeDefined()
  })
  
  test('check is handler has been called', () => {
    const { getByDisplayValue } = render(<InputGroup {...props} />)
    const input = getByDisplayValue(startQuantity)

    fireEvent.change(input, { target: { value: '11' } })
    expect(props.inputHandler).toHaveBeenCalled()

    fireEvent.change(input, { target: { value: '12' } })
    expect(props.inputHandler).toHaveBeenCalledTimes(2)
  })

  test('check is increaseHandler works', () => {
    let quantity = 10
    const increaseHandler = jest.fn().mockImplementation(() => quantity += 1)
    const inputHandler = jest.fn()
    const decreaseHandler = jest.fn()
    
    const { getByDisplayValue } = render(<InputGroup
      quantity={quantity}
      inputHandler={inputHandler}
      increaseHandler={increaseHandler}
      decreaseHandler={decreaseHandler}
    />)
    const arrowTop = getByDisplayValue(startQuantity).nextElementSibling.children[0]

    expect(quantity).toBe(10)
    fireEvent.click(arrowTop)
    expect(increaseHandler).toHaveBeenCalled()
    expect(quantity).toBe(11)
  })

  test('check is decreaseHandler works', () => {
    let quantity = 10
    const decreaseHandler = jest.fn().mockImplementation(() => quantity -= 1)
    const increaseHandler = jest.fn()
    const inputHandler = jest.fn()
    
    const { getByDisplayValue } = render(<InputGroup
      quantity={quantity}
      inputHandler={inputHandler}
      increaseHandler={increaseHandler}
      decreaseHandler={decreaseHandler}
    />)
    const arrowDown = getByDisplayValue(startQuantity).nextElementSibling.children[1]
    expect(quantity).toBe(10)
    fireEvent.click(arrowDown)
    expect(decreaseHandler).toHaveBeenCalled()
    expect(quantity).toBe(9)
  })
})