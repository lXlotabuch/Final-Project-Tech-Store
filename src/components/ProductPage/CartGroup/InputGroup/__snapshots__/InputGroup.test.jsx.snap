// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All test for input group InputGroup shapshot test 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ bKNYfC"
  >
    <input
      class="sc-gsTCUz iyxYNu"
      value="10"
    />
    <div
      class="sc-dlfnbm iYsBYk"
    >
      <div
        class="sc-hKgILt bLNjfg"
        id="topArrow"
      >
        <svg
          fill="none"
          height="5"
          viewBox="0 0 6 5"
          width="6"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 3.8999L3 1.8999L5 3.8999"
            stroke="#8C8C8C"
            stroke-linecap="round"
            stroke-width="1.6"
          />
        </svg>
      </div>
      <div
        class="sc-hKgILt bLNjfg"
      >
        <svg
          fill="none"
          height="5"
          viewBox="0 0 6 5"
          width="6"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5 0.820068L3 2.82007L1 0.820068"
            stroke="#8C8C8C"
            stroke-linecap="round"
            stroke-width="1.6"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
